scene/3d/bone_attachment_3d.h

```diff
@@ -38,20 +38,59 @@ class BoneAttachment3D : public Node3D {
 
 	bool bound = false;
 	String bone_name;
+	int bone_idx = -1;
+
+	bool override_pose = false;
+	int override_mode = 0;
+	bool _override_dirty = false;
+
+	enum OVERRIDE_MODES {
+		MODE_GLOBAL_POSE,
+		MODE_LOCAL_POSE,
+		MODE_CUSTOM_POSE
+	};
+
+	bool use_external_skeleton = false;
+	NodePath external_skeleton_node;
+	ObjectID external_skeleton_node_cache;
 
 	void _check_bind();
 	void _check_unbind();
 
+	void _transform_changed();
+	void _update_external_skeleton_cache();
+	Skeleton3D *_get_skeleton3d();
+
 protected:
 	virtual void _validate_property(PropertyInfo &property) const override;
+	bool _get(const StringName &p_path, Variant &r_ret) const;
+	bool _set(const StringName &p_path, const Variant &p_value);
+	void _get_property_list(List<PropertyInfo> *p_list) const;
```

Comment:
Please consider to add `virtual` and `override`, this will allow the compiler to better guide us in case of refactoring:
```c++
	virtual bool _get(const StringName &p_path, Variant &r_ret) const override;
	virtual bool _set(const StringName &p_path, const Variant &p_value) override;
	virtual void _get_property_list(List<PropertyInfo> *p_list) const override;
```